sdks/python/apache_beam/typehints/native_type_compatibility.py

```diff
@@ -84,19 +84,38 @@ def _safe_issubclass(derived, parent):
     issubclass(derived, parent), or False if a TypeError was raised.
   """
   try:
-    return issubclass(derived, parent)
+    if type(derived) == typing._GenericAlias:
+      out = issubclass(derived.__origin__, parent.__origin__)
+    else:
+      out = issubclass(derived, parent)
+    return out
   except TypeError:
     return False
+  except AttributeError:
+    try:
+      out = issubclass(derived, parent)
+      return out
+    except TypeError:
+      return False
 
 
 def _match_issubclass(match_against):
   return lambda user_type: _safe_issubclass(user_type, match_against)
 
 
+def _matcher_(derived, parent):
+  """ Follow structure of _safe_issubclass
+  but uses __orign__ due to updates in typing package of python3"""
+  try:
+    return derived.__origin__ is parent
```

Comment:
Do we by  chance need to compare all combinations here?

e.g. type1 is type2 or type1.__origin__ is type2 or type1 is type2.__origin__?

Reply:
Hmm don't know if that is needed. Will do some tests and try it.

Reply:
Same as above, but would be great if you could elaborate on why you think it might be needed? Maybe I am missing something.